File: Data Structures & Algorithms/is-palindrome/submission-2.cs
public class Solution {
    public bool IsPalindrome(string s) {
        int left = 0;
        int right = s.Length - 1;
        while (left < right) {
            while (left < right && !Char.IsLetterOrDigit(s[left])) left += 1;
            while (right > left && !Char.IsLetterOrDigit(s[right])) right -= 1;
            if (Char.ToLower(s[left]) != Char.ToLower(s[right])) return false;
            left++; right--;
        }
        return true;
    }
}

File: Data Structures & Algorithms/binary-search/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1
        
        while left < right:
            mid = (left + right) // 2
            if nums[mid] < target:
                left = mid + 1
            elif nums[mid] > target:
                right = mid
            else:
                return mid
        
        if left == right and nums[left] == target:
            return left

        return -1
            